practiceapp/src/test/java/com/practiceapp/practiceapp/controller/post/CreatePostControllerTest.java

```diff
@@ -0,0 +1,86 @@
+package com.practiceapp.practiceapp.controller.post;
+
+
+import com.practiceapp.practiceapp.entity.CommunityEntity;
+import com.practiceapp.practiceapp.entity.PostEntity;
+import com.practiceapp.practiceapp.entity.UserEntity;
+import com.practiceapp.practiceapp.repository.PostRepository;
+import com.practiceapp.practiceapp.service.PostService;
+import org.junit.Test;
+import static org.mockito.Mockito.*;
+
+
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class CreatePostControllerTest {
+
+    //It allows us to execute requests against the test context.
+    @Autowired
+    private MockMvc mockMvc;
+
+    //Fields for mock post
+    Date mock_date = new Date();
+
+    UserEntity mock_user = new UserEntity();
+
+    List<CommunityEntity> communities = new ArrayList<>(List.of(new CommunityEntity()));
+
+    PostEntity mock_post = new PostEntity("123","this is a mock post",mock_date,"istanbul",mock_user,communities);
+
+    String examplePost = "{\"post_id\":\"123\",\"text\":\"this is a mock post\"}";
+
+
+
+
+
+    @Test
+    public void savePostTest() throws Exception {
+
+
+
+        PostService postService = mock(PostService.class);  //mock the post service
+        PostRepository postRepository = mock(PostRepository.class);   //mock post repository
+
+        //mock the behavior of service entity and repository
+        when(postService.savePost(any(PostEntity.class))).thenReturn(1);
+        when(postRepository.save(any(PostEntity.class))).thenReturn(mock_post);
```

Comment:
The controller communicates with the service, not the repository.  So, I'm not sure mocking repo is necessary.

Reply:
Yes you are right. I updated that.